OrbitQt/opengldetect.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "opengldetect.h"
+
+#include <QOffscreenSurface>
+#include <QOpenGLContext>
+
+namespace OrbitGl {
+
+// Detects the supported version of Desktop OpenGL by requesting the most
+// recent version of OpenGL and checking what the system could provide.
+// OpenGL ES is automatically ignored.
+std::optional<OpenGlVersion> DetectOpenGlVersion() {
+  QSurfaceFormat format{};
+  format.setRenderableType(QSurfaceFormat::OpenGL);
+  format.setVersion(4, 6);
```

Comment:
What are 4 and 6 here?

Reply:
This refers to OpenGL 4.6 which is AFAIK the latest version.

Reply:
If a user has OpenGL support but not to the very latest version, will the test will fail?

Reply:
No, it won't. `format` defines a wishlist of features, OpenGL-versions, OpenGL-profiles, etc. which we want to request.

After the context was created we can check whether we got all the features we requested or not. In line 37/38 I call `glContext.format()` which returns a QSurfaceFormat object which represents the available feature set.

Reply:
When I try this on my linux machine I get OpenGL 3.3, on Windows I get 4.6. And on a RDP-only machine this fails.

Reply:
Ok great, thanks for the clarification.